### PR TITLE
[ISSUES 111]Flow rule cache/check

### PR DESCRIPTION
**Pull request by sanxun0325**
### Describe what this PR does / why we need it
In order to avoid the meaningless updates to the property, downstream of property manager should
cache last update value to check consistency.

### Does this pull request fix one issue?

Fix:https://github.com/alibaba/sentinel-golang/issues/111
### Describe how you did it
Using reflect.deepequal,maybe we need to think about performance

### Describe how to verify it
ut

### Special notes for reviews

### CODE REVIEW

File: core/flow/rule_manager.go [hunk 1]
```diff
@@ -88,12 +89,16 @@ func onRuleUpdate(rules []*Rule) (err error) {
...
 	tcMap = m
+	currentRules = rules
 	return nil
 }
 
 // LoadRules loads the given flow rules to the rule manager, while all previous rules will be replaced.
 func LoadRules(rules []*Rule) (bool, error) {
 	// TODO: rethink the design
+	if isEqual := reflect.DeepEqual(currentRules, rules); isEqual {
+		return isEqual, errors.New("The current Rules is the same as the rules to be loaded")
```
Comment (louyuting) [inline core/flow/rule_manager.go:100]: it should return `false, nil`

**Comment by louyuting** [comment]
@sczyh30  How do you think about this design?

File: core/flow/rule_manager.go (hunk 1 above)
Comment (sanxun0325) [reply core/flow/rule_manager.go]: updated

**Comment by louyuting** [comment]
Could you please solve conflicts?

**Pull request closed without merging**